### PR TITLE
Break long `exports` into multiple lines.

### PR DESCRIPTION
**Pull request by demoneaux**
Ref #19.

Currently this PR (https://github.com/jlongster/prettier/pull/104/commits/244c810bd0ef7a41fb76624fb9352652fcfae32c) is based on #101, so you might have to merge that in first, otherwise there has to be a lot of rebasing :|

### CODE REVIEW

File: tests/prettier/exports.js [hunk 1]
```diff
@@ -0,0 +1 @@
+export { value1, value2 as value2_renamed, value3, value4 as value4_renamed, value5 } from "exports";
```
Comment (demoneaux) [inline tests/prettier/exports.js]: I added a new test file with longer exports for easier reference.

File: src/printer.js [hunk 2]
```diff
@@ -1828,15 +1832,33 @@ function printExportDeclaration(path, options, print) {
     } else {
       parts.push(
         decl.exportKind === "type" ? "type " : "",
-        shouldPrintSpaces ? "{ " : "{",
-        join(", ", path.map(print, "specifiers")),
-        shouldPrintSpaces ? " }" : "}"
+        conditionalGroup([
```
Comment (jlongster) [inline src/printer.js]: You don't need `conditionalGroup` here. You only need that in very advanced cases. Not that it's only used in 3 or 4 other places.

This one can be broken like an object with a normal group

```js
group(concat([
  "{",
  indent(
    options.tabWidth,
    concat([
      hardline,
      join(concat([",", options.bracketSpacing ? line : softline]),
           path.map(print, "specifiers"))
    ])
  ),
  options.bracketSpacing ? line : softline,
  "}"
]))
```

That code may not by fully correct, but it's mostly there.

**Comment by jlongster** [comment]
@demoneaux Any chance you could convert this to use `group` instead?

**Comment by jlongster** [comment]
Superseded by https://github.com/jlongster/prettier/pull/156, I went ahead and made those changes so I can get it in. Thanks for starting the work!

**Comment by demoneaux** [comment]
Thanks for providing advice and accepting my fixes! Looking forward to contributing more to this project.

**Pull request closed without merging**